src/Pelagos/Bundle/AppBundle/Rabbit/Consumer/DoiConsumer.php

```diff
@@ -408,10 +408,11 @@ private function getDoiStatus($dataset, $doiStatus)
         } elseif ($dataset->getDatasetStatus() === Dataset::DATASET_STATUS_ACCEPTED and
             $doiStatus === DOI::STATUS_PUBLIC and $restriction === DatasetSubmission::RESTRICTION_RESTRICTED) {
             $status = DOI::STATUS_UNAVAILABLE;
-        } elseif ($dataset->getDatasetStatus() === Dataset::DATASET_STATUS_ACCEPTED and
-            in_array($doiStatus, [DOI::STATUS_RESERVED, DOI::STATUS_UNAVAILABLE]) and
-            $restriction === DatasetSubmission::RESTRICTION_NONE) {
-            $status = DOI::STATUS_PUBLIC;
+        } elseif ($dataset->getDatasetStatus() === Dataset::DATASET_STATUS_ACCEPTED
+            and $restriction === DatasetSubmission::RESTRICTION_NONE) {
+            if ($doiStatus === DOI::STATUS_RESERVED or (preg_match('/^unavailable (\w+)/i', $doiStatus) !== false)) {
```

Comment:
I'm not sure this regex works? You can simply just put the work "unavailable" that way if it appears anywhere it will match. If you want to match it only at the beginning of the phrase use "^unavailable".

Also preg_match return an integer (1 for matches, 0 for no matches, OR false Boolean for an error).

Reply:
Made the change 👍